### ISSUE TITLE
Jail not banning, cant find cause

### ISSUE THREAD

**Issue by corbinjurgens** [fail2ban/fail2ban#2856]
### Environment:

- Fail2Ban v0.11.1. Initially was on 0.9.7 but updated to try fix this issue but didn't help.
- OS, including release name/version: CentOS Linux release 7.7.1908 (Core)
- [x] Fail2Ban installed via OS/distribution mechanisms
- [x] You have not applied any additional foreign patches to the codebase
- [x] Some customizations were done to the configuration

### The issue:

Followed [this](https://rtcamp.com/tutorials/nginx/fail2ban/) guide, and no matter how much I try I cannot get the jail to ban naturally. Manual banning, and regex confirmation is ok.

jail.local
```
[nginx-req-limit]
enabled = true
filter = nginx-custom
action = iptables-multiport[name=ReqLimit, port="http,https", protocol=tcp]
logpath = /var/log/nginx*/*error*.log
findtime = 3600
bantime = 7200
maxretry = 3
backend = systemd
```

filter nginx-custom.conf
```
# Fail2Ban configuration file
#
# supports: ngx_http_limit_req_module module

[Definition]

failregex = limiting requests, excess:.* by zone.*client: <HOST>
datepattern = ^%%Y/%%m/%%d\s+%%H:%%M:%%S
# Option: ignoreregex
# Notes.: regex to ignore. If this regex matches, the line is ignored.
# Values: TEXT
#

ignoreregex =
```

fail2ban-regex result
sudo fail2ban-regex /var/log/nginx/*error.log /etc/fail2ban/filter.d/nginx-custom.conf
```


Running tests
=============

Use   failregex filter file : nginx-custom, basedir: /etc/fail2ban
Use      datepattern : ^Year/Month/Day\s+24hour:Minute:Second
Use         log file : /var/log/nginx/error.log
Use         encoding : ANSI_X3.4-1968


Results
=======

Failregex: 461 total
|-  #) [# of hits] regular expression
|   1) [461] limiting requests, excess:.* by zone.*client: <HOST>
`-

Ignoreregex: 0 total

Date template hits:
|- [# of hits] date format
|  [503] ^Year/Month/Day\s+24hour:Minute:Second
`-

Lines: 503 lines, 0 ignored, 461 matched, 42 missed
[processed in 0.06 sec]

Missed line(s): too many to print.  Use --print-all-missed to print all 42 lines
```

Nginx log excerpt from /var/log/nginx/error.log
```
2020/10/20 12:45:56 [error] 24717#24717: *17991 limiting requests, excess: 5.294 by zone "one", client: 66.249.68.11, server: xx.com, request: "GET xx HTTP/1.1", host: "xx.com"
2020/10/20 12:45:57 [error] 24717#24717: *17991 limiting requests, excess: 5.126 by zone "one", client: 66.249.68.11, server: xx.com, request: "GET xx HTTP/1.1", host: "xx.com"
2020/10/20 12:45:57 [error] 24717#24717: *17991 limiting requests, excess: 5.625 by zone "one", client: 66.249.68.11, server: xx.com, request: "GET xx HTTP/1.1", host: "xx.com"
2020/10/20 12:45:57 [error] 24717#24717: *17991 limiting requests, excess: 5.290 by zone "one", client: 66.249.68.11, server: xx.com, request: "GET xx HTTP/1.1", host: "xx.com"
2020/10/20 12:46:30 [error] 24717#24717: *17991 limiting requests, excess: 5.545 by zone "one", client: 66.249.68.11, server: xx, request: "GET xx HTTP/1.1", host: "xx.com"
2020/10/20 12:46:30 [error] 24717#24717: *17991 limiting requests, excess: 5.374 by zone "one", client: 66.249.68.11, server: xx.com, request: "GET xx HTTP/1.1", host: "xx.com"
2020/10/20 12:46:30 [error] 24717#24717: *17991 limiting requests, excess: 5.202 by zone "one", client: 66.249.68.11, server: xx.com, request: "GET xx HTTP/1.1", host: "xx.com"
2020/10/20 12:46:31 [error] 24717#24717: *17991 limiting requests, excess: 5.315 by zone "one", client: 66.249.68.11, server: xx.com, request: "GET xx HTTP/1.1", host: "xx.com"
```
sudo fail2ban-client status
```
Status
|- Number of jail:      2
`- Jail list:   nginx-req-limit, sshd
```

sudo fail2ban-client status nginx-req-limit
No matter how much I test, and force Nginx limiting requests logs (eg 10 notices within a minute) the jail has absolutely no activity.
Only a manual ban via "sudo fail2ban-client set nginx-req-limit banip 100.100.100.100" is showing
```
Status for the jail: nginx-req-limit
|- Filter
|  |- Currently failed: 0
|  |- Total failed:     0
|  `- Journal matches:
`- Actions
   |- Currently banned: 1
   |- Total banned:     1
   `- Banned IP list:   100.100.100.100
```

#### Steps to reproduce
This is the guide I followed 
https://rtcamp.com/tutorials/nginx/fail2ban/

#### Expected behavior
I should be seeing IPs becoming banned that appear 3 times in the nginx log. 

#### Observed behavior
Absolutely nothing is happening no matter how many times a particular IP appears in the nginx log. The jail is active but it is not doing anything. . This makes me believe somehow the accessing of the nginx log is the point of issue, as manual ban is ok, and regex test is ok

#### Any additional information
I am also using sshd jail, which is working as intended, accessing the /var/log/secure log and banning without issue

#### Config from fail2ban-client -d
```
['set', 'syslogsocket', 'auto']
['set', 'loglevel', 'INFO']
['set', 'logtarget', '/var/log/fail2ban.log']
['set', 'dbfile', '/var/lib/fail2ban/fail2ban.sqlite3']
['set', 'dbmaxmatches', 10]
['set', 'dbpurgeage', '1d']
['add', 'sshd', 'systemd']
['set', 'sshd', 'usedns', 'warn']
['set', 'sshd', 'prefregex', '^<F-MLFID>\\s*(?:\\S+\\s+)?(?:sshd(?:\\[\\d+\\])?:?\\s+)?(?:kernel:\\s?\\[ *\\d+\\.\\d+\\]:?\\s+)?</F-MLFID>(?:(?:error|fatal): (?:PAM: )?)?<F-CONTENT>.+</F-CONTENT>$']
['set', 'sshd', 'maxlines', 1]
['multi-set', 'sshd', 'addfailregex', ['^[aA]uthentication (?:failure|error|failed) for <F-USER>.*</F-USER> from <HOST>( via \\S+)?(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*$', '^User not known to the underlying authentication module for <F-USER>.*</F-USER> from <HOST>(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*$', '^Failed publickey for invalid user <F-USER>(?P<cond_user>\\S+)|(?:(?! from ).)*?</F-USER> from <HOST>(?: (?:port \\d+|on \\S+)){0,2}(?: ssh\\d*)?(?(cond_user): |(?:(?:(?! from ).)*)$)', '^Failed \\b(?!publickey)\\S+ for (?P<cond_inv>invalid user )?<F-USER>(?P<cond_user>\\S+)|(?(cond_inv)(?:(?! from ).)*?|[^:]+)</F-USER> from <HOST>(?: (?:port \\d+|on \\S+)){0,2}(?: ssh\\d*)?(?(cond_user): |(?:(?:(?! from ).)*)$)', '^<F-USER>ROOT</F-USER> LOGIN REFUSED FROM <HOST>', '^[iI](?:llegal|nvalid) user <F-USER>.*?</F-USER> from <HOST>(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*$', '^User <F-USER>.+</F-USER> from <HOST> not allowed because not listed in AllowUsers(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*$', '^User <F-USER>.+</F-USER> from <HOST> not allowed because listed in DenyUsers(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*$', '^User <F-USER>.+</F-USER> from <HOST> not allowed because not in any group(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*$', '^refused connect from \\S+ \\(<HOST>\\)', '^Received <F-MLFFORGET>disconnect</F-MLFFORGET> from <HOST>(?: (?:port \\d+|on \\S+)){0,2}:\\s*3: .*: Auth fail(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*$', '^User <F-USER>.+</F-USER> from <HOST> not allowed because a group is listed in DenyGroups(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*$', "^User <F-USER>.+</F-USER> from <HOST> not allowed because none of user's groups are listed in AllowGroups(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*$", '^<F-NOFAIL>pam_[a-z]+\\(sshd:auth\\):\\s+authentication failure;</F-NOFAIL>(?:\\s+(?:(?:logname|e?uid|tty)=\\S*)){0,4}\\s+ruser=<F-ALT_USER>\\S*</F-ALT_USER>\\s+rhost=<HOST>(?:\\s+user=<F-USER>\\S*</F-USER>)?(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*$', '^(error: )?maximum authentication attempts exceeded for <F-USER>.*</F-USER> from <HOST>(?: (?:port \\d+|on \\S+)){0,2}(?: ssh\\d*)?(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*$', '^User <F-USER>.+</F-USER> not allowed because account is locked(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*', '^<F-MLFFORGET>Disconnecting</F-MLFFORGET>(?: from)?(?: (?:invalid|authenticating)) user <F-USER>\\S+</F-USER> <HOST>(?: (?:port \\d+|on \\S+)){0,2}:\\s*Change of username or service not allowed:\\s*.*\\[preauth\\]\\s*$', '^<F-MLFFORGET>Disconnecting</F-MLFFORGET>: Too many authentication failures(?: for <F-USER>.+?</F-USER>)?(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*$', '^<F-NOFAIL>Received <F-MLFFORGET>disconnect</F-MLFFORGET></F-NOFAIL> from <HOST>(?: (?:port \\d+|on \\S+)){0,2}:\\s*11:', '^<F-NOFAIL><F-MLFFORGET>(Connection closed|Disconnected)</F-MLFFORGET></F-NOFAIL> (?:by|from)(?: (?:invalid|authenticating) user <F-USER>\\S+|.+?</F-USER>)? <HOST>(?:(?: (?:port \\d+|on \\S+|\\[preauth\\])){0,3}\\s*|\\s*)$', '^<F-MLFFORGET><F-MLFGAINED>Accepted \\w+</F-MLFGAINED></F-MLFFORGET> for <F-USER>\\S+</F-USER> from <HOST>(?:\\s|$)', '^<F-NOFAIL>Connection from</F-NOFAIL> <HOST>']]
['set', 'sshd', 'datepattern', '{^LN-BEG}']
['set', 'sshd', 'addjournalmatch', '_SYSTEMD_UNIT=sshd.service', '+', '_COMM=sshd']
['set', 'sshd', 'maxretry', 1]
['set', 'sshd', 'addignoreip', '127.0.0.1/8', 'hiddenip/32', 'hiddenip/32']
['set', 'sshd', 'logencoding', 'auto']
['set', 'sshd', 'ignorecommand', '']
['set', 'sshd', 'findtime', '43200']
['set', 'sshd', 'bantime', '43200']
['set', 'sshd', 'addaction', 'firewallcmd-rich-rules']
['multi-set', 'sshd', 'action', 'firewallcmd-rich-rules', [['actionunban', 'ports="ssh"; for p in $(echo $ports | tr ", " " "); do firewall-cmd --remove-rich-rule="rule family=\'<family>\' source address=\'<ip>\' port port=\'$p\' protocol=\'all\' reject type=\'<rejecttype>\'"; done'], ['actionstop', ''], ['actionstart', ''], ['actionban', 'ports="ssh"; for p in $(echo $ports | tr ", " " "); do firewall-cmd --add-rich-rule="rule family=\'<family>\' source address=\'<ip>\' port port=\'$p\' protocol=\'all\' reject type=\'<rejecttype>\'"; done'], ['actioncheck', ''], ['protocol', 'all'], ['family', 'ipv4'], ['port', 'ssh'], ['chain', 'INPUT'], ['service', 'ssh'], ['actiontype', '<multiport>'], ['actname', 'firewallcmd-rich-rules'], ['zone', 'public'], ['bantime', '43200'], ['rich-blocktype', "reject type='<rejecttype>'"], ['rejecttype', 'icmp-port-unreachable'], ['rejecttype?family=inet6', 'icmp6-port-unreachable'], ['blocktype', 'REJECT --reject-with <rejecttype>'], ['name', 'sshd'], ['family?family=inet6', 'ipv6']]]
['add', 'nginx-req-limit', 'systemd']
['set', 'nginx-req-limit', 'usedns', 'warn']
['set', 'nginx-req-limit', 'addfailregex', 'limiting requests, excess:.* by zone.*client: <HOST>']
['set', 'nginx-req-limit', 'datepattern', '^%Y/%m/%d\\s+%H:%M:%S']
['set', 'nginx-req-limit', 'maxretry', 3]
['set', 'nginx-req-limit', 'addignoreip', '127.0.0.1/8']
['set', 'nginx-req-limit', 'logencoding', 'auto']
['set', 'nginx-req-limit', 'ignorecommand', '']
['set', 'nginx-req-limit', 'findtime', '3600']
['set', 'nginx-req-limit', 'bantime', '7200']
['set', 'nginx-req-limit', 'addaction', 'iptables-multiport']
['multi-set', 'nginx-req-limit', 'action', 'iptables-multiport', [['actionunban', '<iptables> -D f2b-ReqLimit -s <ip> -j <blocktype>'], ['actionflush', '<iptables> -F f2b-ReqLimit'], ['actionstop', '<iptables> -D INPUT -p tcp -m multiport --dports http,https -j f2b-ReqLimit\n<iptables> -F f2b-ReqLimit\n<iptables> -X f2b-ReqLimit'], ['actionstart', '<iptables> -N f2b-ReqLimit\n<iptables> -A f2b-ReqLimit -j RETURN\n<iptables> -I INPUT -p tcp -m multiport --dports http,https -j f2b-ReqLimit'], ['actionban', '<iptables> -I f2b-ReqLimit 1 -s <ip> -j <blocktype>'], ['actioncheck', "<iptables> -n -L INPUT | grep -q 'f2b-ReqLimit[ \\t]'"], ['iptables', 'iptables <lockingopt>'], ['lockingopt', '-w'], ['blocktype?family=inet6', 'REJECT --reject-with icmp6-port-unreachable'], ['protocol', 'tcp'], ['name', 'ReqLimit'], ['chain', 'INPUT'], ['returntype', 'RETURN'], ['iptables?family=inet6', 'ip6tables <lockingopt>'], ['blocktype', 'REJECT --reject-with icmp-port-unreachable'], ['actname', 'iptables-multiport'], ['port', 'http,https']]]
['start', 'sshd']
['start', 'nginx-req-limit']
```

#### Relevant parts of /var/log/fail2ban.log file:
Only info in log I can find is just starting and stopping each time ive been trying to reload. 
```
2020-10-20 11:53:34,277 fail2ban.server         [30162]: DEBUG   Stopping jail nginx-req-limit
2020-10-20 11:53:34,395 fail2ban.actions        [30162]: NOTICE  [nginx-req-limit] Unban 100.100.100.100
2020-10-20 11:53:34,444 fail2ban.filtersystemd  [30162]: DEBUG   nginx-req-limit filter terminated
2020-10-20 11:53:34,723 fail2ban.actions        [30162]: DEBUG   nginx-req-limit: action terminated
2020-10-20 11:53:34,724 fail2ban.jail           [30162]: INFO    Jail 'nginx-req-limit' stopped
2020-10-20 11:53:35,129 fail2ban.jail           [32291]: INFO    Creating new jail 'nginx-req-limit'
2020-10-20 11:53:35,130 fail2ban.jail           [32291]: INFO    Jail 'nginx-req-limit' uses systemd {}
2020-10-20 11:53:36,375 fail2ban.jail           [32291]: INFO    Jail 'nginx-req-limit' started
2020-10-20 11:53:36,456 fail2ban.actions        [32291]: NOTICE  [nginx-req-limit] Ban 100.100.100.100
2020-10-20 12:08:05,024 fail2ban.actions        [32291]: NOTICE  [nginx-req-limit] Unban 100.100.100.100
2020-10-20 12:08:05,887 fail2ban.jail           [32291]: INFO    Jail 'nginx-req-limit' stopped
```

#### Relevant lines from monitored log files in question:
Direct side by side monitor. I forced a few errors to appear in nginx error log and took fail2ban log from that moment
Nginx error.log
```
2020/10/20 13:14:44 [error] 24717#24717: *20635 limiting requests, excess: 5.490 by zone "one", client: myip, server: xx.com, request: "GET /en/guide/ HTTP/1.1", host: "xx.com", referrer: "https://xx.com/en/index"
2020/10/20 13:14:44 [error] 24717#24717: *20635 limiting requests, excess: 5.382 by zone "one", client: myip, server: xx.com, request: "GET /en/guide/ HTTP/1.1", host: "xx.com", referrer: "https://xx.com/en/index"
2020/10/20 13:14:44 [error] 24717#24717: *20635 limiting requests, excess: 5.267 by zone "one", client: myip, server: xx.com, request: "GET /en/guide/ HTTP/1.1", host: "xx.com", referrer: "https://xx.com/en/index"
2020/10/20 13:14:44 [error] 24715#24715: *20637 limiting requests, excess: 5.070 by zone "one", client: myip, server: xx.com, request: "GET /en/guide/ HTTP/1.1", host: "xx.com", referrer: "https://xx.com/en/index"
2020/10/20 13:14:44 [error] 24715#24715: *20637 limiting requests, excess: 5.017 by zone "one", client: myip, server: xx.com, request: "GET /en/guide/ HTTP/1.1", host: "xx.com", referrer: "https://xx.com/en/index"
2020/10/20 13:14:50 [error] 24716#24716: *20651 limiting requests, excess: 5.308 by zone "one", client: myip, server: xx.com, request: "GET /en/guide/ HTTP/1.1", host: "xx.com", referrer: "https://xx.com/en/index"a
```

fail2ban log 
```
As can be seen the IP I tested (hidden as myip) does not appear in this list. Also to note, the IP I manually banned is being processed (auto unban). The only logs appearing are for the sshd jail. 
2020-10-20 13:14:46,504 fail2ban.actions        [5942]: NOTICE  [nginx-req-limit] Unban 100.100.100.100
2020-10-20 13:14:46,505 fail2ban.actions        [5942]: DEBUG   [nginx-req-limit] action 'iptables-multiport': unban 100.100.100.100
2020-10-20 13:14:46,515 fail2ban.utils          [5942]: DEBUG   7f68f83d94b0 -- returned successfully 0
2020-10-20 13:14:46,526 fail2ban.utils          [5942]: DEBUG   7f68d80d9f10 -- returned successfully 0
2020-10-20 13:14:46,526 fail2ban.actions        [5942]: DEBUG   Unbanned 1, 0 ticket(s) in 'nginx-req-limit'
2020-10-20 13:14:47,027 fail2ban.filter         [5942]: DEBUG   Processing line with time:1603167287.03 and ip:208.109.9.14
2020-10-20 13:14:47,029 fail2ban.filter         [5942]: INFO    [sshd] Found 208.109.9.14 - 2020-10-20 13:14:47
2020-10-20 13:14:47,030 fail2ban.failmanager    [5942]: DEBUG   Total # of detected failures: 10552. Current failures from 1 IPs (IP:count): 208.109.9.14:1
2020-10-20 13:14:47,054 fail2ban.filter         [5942]: DEBUG   Processing line with time:1603167287.05 and ip:208.109.9.14
2020-10-20 13:14:47,055 fail2ban.filter         [5942]: INFO    [sshd] Found 208.109.9.14 - 2020-10-20 13:14:47
2020-10-20 13:14:47,055 fail2ban.failmanager    [5942]: DEBUG   Total # of detected failures: 10553. Current failures from 1 IPs (IP:count): 208.109.9.14:1
2020-10-20 13:14:47,254 fail2ban.filter         [5942]: DEBUG   Processing line with time:1603167287.25 and ip:191.234.189.215
2020-10-20 13:14:47,255 fail2ban.filter         [5942]: INFO    [sshd] Found 191.234.189.215 - 2020-10-20 13:14:47
2020-10-20 13:14:47,255 fail2ban.failmanager    [5942]: DEBUG   Total # of detected failures: 10554. Current failures from 1 IPs (IP:count): 191.234.189.215:1
2020-10-20 13:14:47,461 fail2ban.actions        [5942]: WARNING [sshd] 208.109.9.14 already banned
2020-10-20 13:14:47,461 fail2ban.actions        [5942]: WARNING [sshd] 208.109.9.14 already banned
2020-10-20 13:14:47,461 fail2ban.actions        [5942]: WARNING [sshd] 191.234.189.215 already banned
2020-10-20 13:14:48,519 fail2ban.filter         [5942]: DEBUG   Processing line with time:1603167288.52 and ip:140.113.43.34
2020-10-20 13:14:48,520 fail2ban.filter         [5942]: INFO    [sshd] Found 140.113.43.34 - 2020-10-20 13:14:48
2020-10-20 13:14:48,521 fail2ban.failmanager    [5942]: DEBUG   Total # of detected failures: 10555. Current failures from 1 IPs (IP:count): 140.113.43.34:1
2020-10-20 13:14:48,552 fail2ban.filter         [5942]: DEBUG   Processing line with time:1603167288.55 and ip:140.113.43.34
2020-10-20 13:14:48,553 fail2ban.filter         [5942]: INFO    [sshd] Found 140.113.43.34 - 2020-10-20 13:14:48
2020-10-20 13:14:48,553 fail2ban.failmanager    [5942]: DEBUG   Total # of detected failures: 10556. Current failures from 1 IPs (IP:count): 140.113.43.34:1
2020-10-20 13:14:48,864 fail2ban.actions        [5942]: WARNING [sshd] 140.113.43.34 already banned
2020-10-20 13:14:48,864 fail2ban.actions        [5942]: WARNING [sshd] 140.113.43.34 already banned
2020-10-20 13:14:58,668 fail2ban.filter         [5942]: DEBUG   Processing line with time:1603167298.66 and ip:119.45.139.249
2020-10-20 13:14:58,668 fail2ban.filter         [5942]: INFO    [sshd] Found 119.45.139.249 - 2020-10-20 13:14:58
2020-10-20 13:14:58,669 fail2ban.failmanager    [5942]: DEBUG   Total # of detected failures: 10557. Current failures from 1 IPs (IP:count): 119.45.139.249:1
2020-10-20 13:14:58,699 fail2ban.filter         [5942]: DEBUG   Processing line with time:1603167298.7 and ip:119.45.139.249
2020-10-20 13:14:58,700 fail2ban.filter         [5942]: INFO    [sshd] Found 119.45.139.249 - 2020-10-20 13:14:58
2020-10-20 13:14:58,700 fail2ban.failmanager    [5942]: DEBUG   Total # of detected failures: 10558. Current failures from 1 IPs (IP:count): 119.45.139.249:1
2020-10-20 13:14:58,887 fail2ban.actions        [5942]: WARNING [sshd] 119.45.139.249 already banned
2020-10-20 13:14:58,887 fail2ban.actions        [5942]: WARNING [sshd] 119.45.139.249 already banned
```

**Comment by sebres** [comment]
> No matter how much I test, and force Nginx limiting requests logs (eg 10 notices within a minute) the jail has absolutely no activity.

Well you configured `backend = systemd` for the jail, what means that this will monitor systemd-journal and not the log-file (remove it or change it to `backend = auto` either and it should work).

> nginx-custom ...

Newer version of fail2ban has already a filter [filter.d/nginx-limit-req.conf](https://github.com/fail2ban/fail2ban/blob/master/config/filter.d/nginx-limit-req.conf) for that.
<details><summary>test with default nginx-limit-req</summary>

```bash
$ msg='2020/10/20 12:45:56 [error] 24717#24717: *17991 limiting requests, excess: 5.294 by zone "one", client: 192.0.2.1, server: xx.com, request: "GET xx HTTP/1.1", host: "xx.com"'
$ fail2ban-regex "$msg" nginx-limit-req

Running tests
=============

Use   failregex filter file : nginx-limit-req, basedir: /etc/fail2ban
Use      datepattern : Default Detectors
Use      single line : 2020/10/20 12:45:56 [error] 24717#24717: *17991 li...


Results
=======

Failregex: 1 total
|-  #) [# of hits] regular expression
|   1) [1] ^\s*\[[a-z]+\] \d+#\d+: \*\d+ limiting requests, excess: [\d\.]+ by zone "(?:[^"]+)", client: <HOST>,
`-

Ignoreregex: 0 total

Date template hits:
|- [# of hits] date format
|  [1] {^LN-BEG}ExYear(?P<_sep>[-/.])Month(?P=_sep)Day(?:T|  ?)24hour:Minute:Second(?:[.,]Microseconds)?(?:\s*Zone offset)?
`-

Lines: 1 lines, 0 ignored, 1 matched, 0 missed
[processed in 0.00 sec]
```
</details>

> I am also using sshd jail, which is working as intended, accessing the /var/log/secure log and banning without issue

Hmm... Looks not good to me, as I see many of `WARNING [sshd] ... already banned` messages. This means that despite an active ban, the intruder seems to be able to generate a failures yet.
This can mean the banning action does not work, or there is some white-listening rules, or there is some another way to connect sshd (the port is something other as defined in jail), or something similar.

**Comment by corbinjurgens** [comment]
Thank you for your reply.

The backend option is set as it is because I was trying to find the cause and tried many options. In my original tests it was not set. I will try again and let you know the result

**Comment by sebres** [comment]
> As for the already banned messages, I was aware this is strange and tried to make ssh ban all ports but still seem to get the message and gave up

I cannot help here without more details (iptables chains listing, net-filter subsystem you use, etc). 
Please note https://github.com/fail2ban/fail2ban/issues/2698#issuecomment-617336143 or https://github.com/fail2ban/fail2ban/issues/2545#issuecomment-543087971 and below.

**Comment by corbinjurgens** [comment]
I was able to solve the issue.

By making use of the filter [nginx-limit-req.conf](https://github.com/fail2ban/fail2ban/blob/master/config/filter.d/nginx-limit-req.conf) AND changing backend to 'auto' (simply removing seemed to not be enough) the jail is now working correctly. 

As for 'already banned' logs, It seems to only happen when restarting Fail2ban, about 30 messages a second. Once some time passes, the messages slow down completely and I can only see 'found' and 'ban' messages. I imagine it is going back to a point in the past and reanalyzing the logs each time I restart causing the messages. I also changed port and protocol to "all", this may have changes things.

I also switch ssh to iptables (before it was defaulting to firewall-cmd as I can see from my previous config above). The ipconfig appears as 
```
Chain f2b-sshd (1 references)
    pkts      bytes target     prot opt in     out     source               destination
      xx     xx REJECT     all  --  *      *       xxx.xxx.xxx.xxx       0.0.0.0/0            reject-with icmp-port-unreachable
```
...

Thank you for your help.
